File: Programacion II/Practica 2/src/practica1/Punto.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package practica1;
import java.text.DecimalFormat;
/**
 *
 * @author aalex
 */
public class Punto {
    public float x;
    public float y;
    public Punto(float x, float y) {
        this.x = x;
        this.y = y;
    }
    public double[] coordCartesianas() {
        return new double[]{x, y};
    }
    public double[] coordPolares() {
        double r = Math.sqrt(x * x + y * y);
        double theta = Math.atan2(y, x);  // Ángulo en radianes
        return new double[]{r, theta};
    }
    public double getX() { return x; }
    public double getY() { return y; }
    @Override
    public String toString() {
        DecimalFormat df = new DecimalFormat("#.##");
        return "Punto(" + df.format(x) + ", " + df.format(y) + ")";
    }
}

File: Programacion II/Practica 1/Punto.py
import math
class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    def coord_cartesianas(self):
        return (self.x, self.y)
    def coord_polares(self):
        r = math.sqrt(self.x ** 2 + self.y ** 2)
        theta = math.atan2(self.y, self.x)
        return (r, theta)
    def __str__(self):
        return f"Punto({self.x}, {self.y})"
if __name__ == "__main__":
    p = Punto(3, 4)
    print(f"Coordenadas Cartesianas: {p.coord_cartesianas()}")
    print(f"Coordenadas Polares: {p.coord_polares()}")
    print(f"Representación de texto: {p}")
   